### PR TITLE
Updating README and LAYOUT to reflect addition of CONTRIBUTING

### PR DESCRIPTION
**Pull request by gvwilson**
We now have a `CONTRIBUTING.md` file in the `core` branch, so this adds a description of it to `README.md` and `LAYOUT.md`.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    in Scott Chacon's book *Pro Git*.
+
+2.  For our lessons,
+    you should branch from and submit pull requests against the `gh-pages` branch.
+
+3.  When editing lesson pages, you need only commit changes to the Markdown source files.
+
+4.  If you're looking for things to work on,
+    please see the current list of issues
+    for [this repository](https://github.com/swcarpentry/lesson-template/issues),
```
Comment (rgaiacs) [inline CONTRIBUTING.md]: I would suggest to use reference-style links, e.g. `[this repository][github-address]`, to make easy for people find the line they should edit. If this suggestion was accepted will be nice to change `LAYOUT.md` to

```
8.  `CONTRIBUTING.md`: instructions for people who want to contribute.
    This file should be edited to replace references the link associate with `github-address`
    at the end of the file to reference your lesson's repository.
```

**Comment by rgaiacs** [comment]
@gvwilson just a minor suggestion. :+1: to merge.

**Comment by gvwilson** [comment]
Switched to ref-links to make you and @wking happy - should we mandate 
this as style throughout lessons, and get the validator to check for it?

**Comment by rgaiacs** [comment]
> should we mandate this as style throughout lessons

I don't want that we mandate this but I'm OK if you prefer to do so.

> and get the validator to check for it?

If we are going to mandate, yes.

**Comment by rgaiacs** [comment]
I syncronized the `CONTRIBUTING.md` at `core` and `gh-pages` at 7547e80.

**Comment by wking** [comment]
On Fri, Feb 13, 2015 at 11:52:43AM -0800, Raniere Silva wrote:

> > should we mandate this as style throughout lessons
> 
> I don't want that we mandate this but I'm OK if you prefer to do so.

I agree that this should be a per-repo choice and not a SWC-wide
choice.  I don't see why anyone would prefer inline links, but that
doesn't mean that there aren't reasons (or irrational maintainers ;).

> > and get the validator to check for it?
> 
> If we are going to mandate, yes.

I'm not sure how easy such a check would be, but having something that
individual lesson maintainers could configure for their lesson
(e.g. so a check run by Travis for each pull request would inform
contributors of links that didn't match the local style) would be
nice.  I don't think anyone's using CI testing like that now though,
so it's certainly not a high priority.

**Pull request merged**